docs/4.3/aws_oss_guide.md

```diff
@@ -386,12 +379,308 @@ To upgrade to a newer version of Teleport:
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+```
+This will create a `.json` file locally, the below command should then be ran to create the role.
+
+```bash
+$ ROLE_ARN=$(aws iam create-role --role-name teleport-roles --assume-role-policy-document file://teleport_assume_role.json | jq -r '.Role.Arn')
+```
+
+### Create IAM policy granting list-clusters and describe-cluster permissions (optional)
+
+This policy is necessary to create a `kubeconfig` file using the `aws eks update-kubeconfig`
+command. If you have another mechanism to create a kubeconfig file on the instance that runs
+Teleport, this step is not required.
+
+```bash
+cat > teleport_eks_desc_and_list.json << 'EOF'
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+POLICY_ARN=$(aws iam create-policy --policy-name teleport-policy --policy-document file://teleport_eks_desc_and_list.json | jq -r '.Policy.Arn')
+aws iam attach-role-policy --role-name teleport-role --policy-arn $POLICY_ARN
+```
+
+### Update aws-auth configmap
+
+This maps the IAM role **teleport-role** to the Kubernetes group **teleport-group**.
+
+!!! note
+
+    If you used eksctl to create your cluster, you may need to add the mapUsers section to the aws-auth ConfigMap before executing these commands.
+
+    ```
+    eksctl create iamidentitymapping --cluster [cluster-name] --arn arn:aws:iam::[account-id]:role/teleport-role --group teleport-group --username teleport
+    ```
+
+```bash
+ROLE="    - userarn: arn:aws:iam::[account-id]:role/teleport-role\n      username: teleport\n      groups:\n        - teleport-group"
+kubectl get -n kube-system configmap/aws-auth -o yaml | awk "/mapRoles: \|/{print;print \"$ROLE\";next}1" > /tmp/aws-auth-patch.yml
+kubectl patch configmap/aws-auth -n kube-system --patch "$(cat /tmp/aws-auth-patch.yml)"
+```
+
+To check, run  `kubectl describe configmap -n kube-system aws-auth`
+
+
+### Installing Teleport
+#### Create EC2 instance
+Create an EC2 instance using the [Teleport Community AMI](https://aws.amazon.com/marketplace/pp/Gravitational-Teleport-Community-Edition/B07FYTZB9B) on a public subnet in your VPC. Modify the security group associated with that instance to allow port 22 inbound, so you can SSH to the instancea fter it’s running. You will need security group rules to allow access to ports `3080`  and `3022-3026` so that users can access theTeleport server from the Internet.
+
+This  also allows GitHub to post a response back to the Teleport server. You’ll also
+need to open port 80 to allow Let’s Encrypt to complete HTTP validation when issuing
+SSL certificates.
```

Comment:
`s/when using SSL certificates/unless you provide pre-provisioned TLS certificates`